x/circuit/types/genesis.go

```diff
@@ -0,0 +1,25 @@
+package types
+
+import fmt "fmt"
+
+func DefaultGenesisState() *GenesisState {
+	return &GenesisState{}
+}
+
+func (gs *GenesisState) Validate() error {
+
+	for _, accounts := range gs.AccountPermissions {
+		if accounts.Address == "" {
+			return fmt.Errorf("invalid account address: %s", accounts.Address)
+		}
+		if accounts.Permissions == nil {
+			return fmt.Errorf("account has empty permissions, account address: %s", accounts.Address)
+		}
+
+		if accounts.Permissions.Level != Permissions_LEVEL_ALL_MSGS && accounts.Permissions.Level != Permissions_LEVEL_SOME_MSGS && accounts.Permissions.Level != Permissions_LEVEL_SUPER_ADMIN {
```

Comment:
I would add a method for this and use it for level checks, otherwise it is easy to forget to add one.